ampd/src/lib.rs

```diff
@@ -52,20 +52,20 @@ pub async fn run(cfg: Config, state_path: PathBuf) -> Result<(), Error> {
         multisig,
     } = cfg;
 
-    let tm_client =
-        tendermint_rpc::HttpClient::new(tm_jsonrpc.to_string().as_str()).map_err(Error::new)?;
+    let tm_client = tendermint_rpc::HttpClient::new(tm_jsonrpc.to_string().as_str())
+        .change_context(Error::Connection)?;
     let service_client = ServiceClient::connect(tm_grpc.to_string())
         .await
-        .map_err(Error::new)?;
+        .change_context(Error::Connection)?;
     let query_client = QueryClient::connect(tm_grpc.to_string())
         .await
-        .map_err(Error::new)?;
+        .change_context(Error::Connection)?;
     let multisig_client = MultisigClient::connect(tofnd_config.party_uid, tofnd_config.url)
         .await
-        .map_err(Error::new)?;
+        .change_context(Error::Tofnd)?;
```

Comment:
This is also technically a connection issue cuz it's trying to connect to Tofnd via gRPC, no?